pkg/controller/elasticquota.go

```diff
@@ -0,0 +1,329 @@
+func (ctrl *ElasticQuotaController) sync() bool {
+	keyObj, quit := ctrl.eqQueue.Get()
+	if quit {
+		return false
+	}
+	err := func(obj interface{}) error {
+		defer ctrl.eqQueue.Done(obj)
+		key, ok := obj.(string)
+		if !ok {
+			ctrl.eqQueue.Forget(obj)
+			runtime.HandleError(fmt.Errorf("expected string in workqueue but got %#v", keyObj))
+			return nil
+		}
+		if err := ctrl.syncHandler(key); err != nil {
+			return fmt.Errorf("error syncing elastic quota '%s': %s", key, err.Error())
+		}
+		ctrl.eqQueue.Forget(obj)
+		klog.V(5).Infof("Successfully synced '%s'", key)
+		return nil
+	}(keyObj)
+	if err != nil {
+		runtime.HandleError(err)
+		return true
+	}
+	return true
+}
+
+// syncHandler syncs elastic quota and convert status.used
+func (ctrl *ElasticQuotaController) syncHandler(key string) error {
+	// Convert the namespace/name string into a distinct namespace and name
+	namespace, name, err := cache.SplitMetaNamespaceKey(key)
+	if err != nil {
+		runtime.HandleError(fmt.Errorf("invalid resource key: %s", key))
+		return nil
+	}
+	// Get the elastic quota resource with this namespace/name
+	eq, err := ctrl.eqLister.ElasticQuotas(namespace).Get(name)
+	if err != nil {
+		// The elastic quota may no longer exist, in which case we stop processing.
+		if apierrs.IsNotFound(err) {
+			_, err = ctrl.eqClient.SchedulingV1alpha1().ElasticQuotas(namespace).Get(context.TODO(),
+				name, metav1.GetOptions{})
+			if err != nil && apierrs.IsNotFound(err) {
+				// elastic quota was deleted in the meantime, ignore.
+				klog.V(3).Infof("elastic quota has been %q deleted", name)
+				return nil
+			}
+			return nil
+		}
+		runtime.HandleError(fmt.Errorf("failed to get elastic quota by: %s/%s", namespace, name))
+		return err
+	}
+
+	klog.V(5).Infof("Try to process elastic quota: %#v", key)
+	used, err := ctrl.computeNamespacedUsed(namespace, eq)
+	if err != nil {
+		return err
+	}
+	// create a usage object that is based on the elastic quota version that will handle updates
+	// by default, we set used to the current status
+	usage := eq.DeepCopy()
+	usage.Status = schedv1alpha1.ElasticQuotaStatus{
+		Used: used,
+	}
+	// Ignore this loop if the usage value has not changed
+	if apiequality.Semantic.DeepEqual(usage.Status, eq.Status) {
+		return nil
+	}
+	if _, err = ctrl.eqClient.SchedulingV1alpha1().ElasticQuotas(namespace).
+		Update(context.TODO(), usage, metav1.UpdateOptions{}); err != nil {
+		return err
+	}
+	ctrl.recorder.Event(eq, v1.EventTypeNormal, "Synced", "Elastic Quota synced successfully")
+	return nil
+}
+
+func (ctrl *ElasticQuotaController) computeNamespacedUsed(namespace string, eq *schedv1alpha1.ElasticQuota) (v1.ResourceList, error) {
+	used := v1.ResourceList{}
+	pods, err := ctrl.podLister.Pods(namespace).List(labels.NewSelector())
+	if err != nil {
+		return nil, err
+	}
+	for _, p := range pods {
+		// ignore pending pods
+		if p.Status.Phase != v1.PodPending {
```

Comment:
`Succeeded` 、`Failed`、`Unknown`  is also needed to be ignored. ^_^

Reply:
This means that we only consider Pods in the Running state